providers/mlx5/mlx5.c

```diff
@@ -609,16 +610,35 @@ static int mlx5_map_internal_clock(struct mlx5_device *mdev,
 	return 0;
 }
 
+struct ibv_cq_ex *mlx5dv_create_cq_ex(struct ibv_context *context,
+				      struct ibv_cq_init_attr_ex *cq_attr,
+				      struct mlx5dv_cq_init_attr_ex *mlx5_cq_attr)
+{
+	return ibv_vendor_create_cq_ex(context, cq_attr, mlx5_cq_attr);
+}
```

Comment:
I said the lock didn't look needed because there didn't seem to be any reason for it to cover the driver callback.

It looks like the lock only needs to cover refcnt, so you should revise it and just have the one helper function.

Obviously the lock cannot protect against racing create with ibv_destroy_comp_channel, as destroy could happen immediately before the lock is acquired and the channel pointer will be freed. It is an error for the user to allow this to happen.

Ideally refcnt would become a 'kref' like thing using atomics and a lot of this ugly mess would go away including the lock and EBUSY failure...